accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -545,6 +549,38 @@ per
 
 ## Runtime semantics
 
+### Weak and strong semantics
+
+To allow the null safety feature to be rolled out incrementally, we define two
+modes of execution.
+
+**Weak checking** mode largely ignores the nullability of types at runtime, as
+defined below.  Unmigrated programs or programs consisting of a mix of migrated
+and unmigrated code are expected to run without encountering new nullability
+related errors at runtime.  **This mode is unsound** in the sense that variables
+marked as non-nullable may still be null at runtime.
+
+**Strong checking** mode respects the nullability of types at runtime in casts
+and instance checks, as defined below.  Unmigrated programs or programs
+consisting of a mix of migrated and unmigrated code may fail at runtime with new
+cast failures due to the nullability annotations in runtime type objects.
+**This mode is unsound** if any part of the program is still unmigrated.  **This
+mode is sound** if all parts of the program have been migrated.
+
+Weak vs strong runtime checking can be controlled at runtime via the
+`--[no-]runtime-null-safety` flag, where the negated version of the flag implies
```

Comment:
Given these flags are passed to compilers, which are all about running code, I would prefer to simplify the terms by omitting the `runtime-` part. I.e.:

* --null-safety: "strong checking"
* --no-null-safety: "weak checking"

Reply:
Done.